app/repositories/application/ReportingRepoBSONToModelHelper.scala

```diff
@@ -0,0 +1,250 @@
+
+      TestResultsForOnlineTestPassMarkReportItem(behaviouralTestResult, situationalTestResult, etrayTestResult, videoInterviewResults)
+  }
+
+  private[application] def toPhase1TestResults(testGroupsDoc: Option[BSONDocument]) = {
+    val phase1Doc = testGroupsDoc.flatMap(_.getAs[BSONDocument](Phase.PHASE1))
+    val phase1TestProfile = Phase1TestProfile.bsonHandler.read(phase1Doc.get)
+
+    val situationalScheduleId = cubiksGatewayConfig.phase1Tests.scheduleIds("sjq")
+    val behaviouralScheduleId = cubiksGatewayConfig.phase1Tests.scheduleIds("bq")
+
+    def getTestResult(phase1TestProfile: Phase1TestProfile, scheduleId: Int) = {
+      phase1TestProfile.activeTests.find(_.scheduleId == scheduleId).flatMap { phase1Test =>
+        phase1Test.testResult.map { tr => toTestResult(tr) }
+      }
+    }
+    (getTestResult(phase1TestProfile, behaviouralScheduleId), getTestResult(phase1TestProfile, situationalScheduleId))
+  }
+
+  private[application] def toPhase2TestResults(applicationId: String, testGroupsDoc: Option[BSONDocument]) = {
+    val phase2DocOpt = testGroupsDoc.flatMap(_.getAs[BSONDocument](Phase.PHASE2))
+    phase2DocOpt.flatMap { phase2Doc =>
+      val phase2TestProfile = Phase2TestGroup.bsonHandler.read(phase2Doc)
+      phase2TestProfile.activeTests.size match {
+        case 1 => phase2TestProfile.activeTests.head.testResult.map { tr => toTestResult(tr) }
+        case 0 => None
+        case s if s > 1 =>
+          Logger.error(s"There are $s active tests which is invalid for application id [$applicationId]")
+          None
+      }
+    }
+  }
+
+  private[application] def toPhase3TestResults(testGroupsDoc: Option[BSONDocument]): Option[VideoInterviewTestResult] = {
+    val reviewedDocOpt = testGroupsDoc.flatMap(_.getAs[BSONDocument](Phase.PHASE3))
+      .flatMap(_.getAs[BSONArray]("tests")).flatMap(_.getAs[BSONDocument](0))
+      .flatMap(_.getAs[BSONDocument]("callbacks")).flatMap(_.getAs[List[BSONDocument]]("reviewed"))
+
+    val reviewed = reviewedDocOpt.map (_.map(ReviewedCallbackRequest.bsonHandler.read(_)))
+    val latestReviewedOpt = reviewed.map ( _.sortWith { (r1, r2) => r1.received.isAfter(r2.received)}).flatMap(_.headOption)
+
+    latestReviewedOpt.map { latestReviewed =>
+      VideoInterviewTestResult(
+        toVideoInterviewQuestionTestResult(latestReviewed.latestReviewer.question1),
+        toVideoInterviewQuestionTestResult(latestReviewed.latestReviewer.question2),
+        toVideoInterviewQuestionTestResult(latestReviewed.latestReviewer.question3),
+        toVideoInterviewQuestionTestResult(latestReviewed.latestReviewer.question4),
+        toVideoInterviewQuestionTestResult(latestReviewed.latestReviewer.question5),
+        toVideoInterviewQuestionTestResult(latestReviewed.latestReviewer.question6),
+        toVideoInterviewQuestionTestResult(latestReviewed.latestReviewer.question7),
+        toVideoInterviewQuestionTestResult(latestReviewed.latestReviewer.question8),
```

Comment:
Say if one of the user has just received test results for few questions and if some admin user tries to download the report ... do you think it might result in some exceptions?? How does the code behaves in general